### PR TITLE
Improve logging for boolean Service Annotations

### PR DESCRIPTION
**Pull request by FavourEva**
**What this PR does / why we need it**:
Improve logging for bool service annotation

**Which issue this PR fixes(if applicable)**:
fixes #

**Special notes for reviewers**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/openstack/loadbalancer.go [hunk 1]
```diff
@@ -687,13 +688,14 @@ func getBoolFromServiceAnnotation(service *corev1.Service, annotationKey string,
...
 		default:
+			klog.Error("Invalid Service Annotation: %v = %v (falling back to default setting: %v)", annotationKey, annotationValue, defaultSetting)
 			returnValue = defaultSetting
 		}
 
 		klog.V(4).Infof("Found a Service Annotation: %v = %v", annotationKey, returnValue)
 		return returnValue
 	}
-	klog.V(4).Infof("Could not find a Service Annotation; falling back to default setting: %v = %v", annotationKey, defaultSetting)
+	klog.V(4).Infof("Found a non-boolean Service Annotation; falling back to default setting: %v = %v", annotationKey, defaultSetting)
```
Comment (sakshi-1505) [inline pkg/openstack/loadbalancer.go:698]: I don't think it's the right place for non-boolean, if the func execution ever reaches here. It means that we couldn't find any service annotation for the following key. The log message currently indicates that if found an annotation when it didn't.

File: pkg/openstack/loadbalancer.go [hunk 2]
```diff
@@ -687,13 +688,14 @@ func getBoolFromServiceAnnotation(service *corev1.Service, annotationKey string,
 		case "false":
 			returnValue = false
 		default:
+			klog.Error("Invalid Service Annotation: %v = %v (falling back to default setting: %v)", annotationKey, annotationValue, defaultSetting)
```
Comment (sakshi-1505) [inline pkg/openstack/loadbalancer.go:691]: Here you should log that, `found a non-boolean annotation`

**Review by sakshi-1505 — CHANGES_REQUESTED** [review]

File: pkg/openstack/loadbalancer.go [hunk 3]
```diff
@@ -687,6 +688,7 @@ func getBoolFromServiceAnnotation(service *corev1.Service, annotationKey string,
 		case "false":
 			returnValue = false
 		default:
+			klog.Error("Found a non-boolean Service Annotation: %v = %v (falling back to default setting: %v)", annotationKey, annotationValue, defaultSetting)
```
Comment (pierreprinetti) [inline pkg/openstack/loadbalancer.go:691]: Good! Now return `defaultSetting` after this line.

Comment (FavourEva) [reply pkg/openstack/loadbalancer.go:691]: > Good! Now return `defaultSetting` after this line.

Done! Is it  cool now? @pierreprinetti

Comment (pierreprinetti) [reply pkg/openstack/loadbalancer.go:691]: Perfect! Thanks.

**Review by pierreprinetti — APPROVED** [review]
/lgtm

**Comment by pierreprinetti** [comment]
/lgtm cancel

Wait. I am not sure that `klog.Error` is what we want here. @jichenjc @kayrus what would be the appropriate logging level here for consistency with the rest of the logic?

**Comment by FavourEva** [comment]
> /lgtm cancel
> 
> Wait. I am not sure that `klog.Error` is what we want here. @jichenjc @kayrus what would be the appropriate logging level here for consistency with the rest of the logic?

`klog.warningf`?

**Comment by jichenjc** [comment]
are we good enough to go or need fix the log warning suggestion?

**Comment by FavourEva** [comment]
> are we good enough to go or need fix the log warning suggestion?

We actually need a suggestion on what the appropriate logging level should be or is `klog.Error` what we want there? @jichenjc

**Comment by jichenjc** [comment]
I can't find where's orignial info but my understanding is 

Log.info => general info for reference
Log.warning => warning, potentially lead to error in the future or something worthy attention
Log.error => definitely something wrong and already cause issues

**Comment by FavourEva** [comment]
> I can't find where's orignial guideline from golang/k8s cmmunity but my understanding is
> 
> Log.info => general info for reference Log.warning => warning, potentially lead to error in the future or something worthy attention Log.error => definitely something wrong and already cause issues

I guess the logging level should align with the requirements and severity of the situation. Since falling back to default setting is an expected behaviour, I think using a lower log level `klog.Info` would be better. What do you think @jichenjc

**Comment by jichenjc** [comment]
>>>Since falling back to default setting is an expected behaviour

if it is fall back to default (normal way) then yes, Info is better

**Comment by FavourEva** [comment]
> /lgtm cancel
> 
> Wait. I am not sure that `klog.Error` is what we want here. @jichenjc @kayrus what would be the appropriate logging level here for consistency with the rest of the logic?

Resolved! @pierreprinetti Please review

**Review by pierreprinetti — APPROVED** [review]
/lgtm

**Comment by pierreprinetti** [comment]
/assign jichenjc
when you have time :pray:

**Comment by FavourEva** [comment]
@dulek @jichenjc Could you please consider merging my PR when you have a moment? I believe its ready. Thanks a lot

**Comment by jichenjc** [comment]
/approve 

sorry for the delay :)

**Comment by dulek** [comment]
This is a neat improvement, thank you!

**Pull request merged**